src/modules/products/controller/productController.ts

```diff
@@ -0,0 +1,112 @@
+import { Request, Response } from 'express';
+import ProductDao from '../dao/ProductDao';
+import ProductModel from "../model/ProductModel";
+import config from "../../../config";
+
+/**
+ * Creates a new product.
+ * @param req - The request object.
+ * @param res - The response object.
+ */
+
+export const createProduct =  async (req: any, res: Response) => {
+    try {
+        const productDao = new ProductDao(ProductModel);
+        if(req.files && req.files.length > 0) {
+            // Assert the type of req.files to Express.Multer.File[]
+            const files: Express.MulterS3.File[] = req.files;
+            const filePaths: string[] = files?.map(file => ( file.key )) || [];
+            // Prepare product data including file paths
+            const productObject = JSON.parse(req.body.data);
+            const productData = {
+                ...productObject,
+                images: [...filePaths, ...productObject.images],
+            };
+            const product = await productDao.createProduct(productData);
+            res.status(201).json(product);
+        } else {
+            const product = await productDao.createProduct(req.body);
+            res.status(201).json(product);
+        }
+
+    } catch (error) {
+      res.status(500).json({ message: 'Error creating product', error });
+    }
+};
+
+/**
+ * Finds a product by its ID.
+ * @param req - The request object.
+ * @param res - The response object.
+ */
+export const getProductById = async (req: Request, res: Response): Promise<void> => {
+    try {
+        const productDao = new ProductDao(ProductModel)
+        const product = await productDao.findProductById(req.params.id);
+        if (product) {
+            product.imageUrls = [];
+            product.images.map((image: any) => {
```

Comment:
question: type of image is String? maybe could escape from any

Reply:
ec68572fde36832e71effb99ca4101be308ff384